File: wrappers/dotnet/indy-sdk-dotnet/WalletApi/WrongWalletForPoolException.cs
namespace Hyperledger.Indy.WalletApi
{
    /// <summary>
    /// Exception thrown when attempting to use a wallet with the wrong pool.
    /// </summary>
    public class WrongWalletForPoolException : IndyException
    {
        const string message = "The wallet specified is not compatible with the open pool.";

        /// <summary>
        /// Initializes a new WrongWalletForPoolException.
        /// </summary>
        internal WrongWalletForPoolException() : base(message, (int)ErrorCode.WalletIncompatiblePoolError)
        {

        }
    }

}

File: wrappers/android/src/main/java/org/hyperledger/indy/sdk/wallet/WrongWalletForPoolException.java
package org.hyperledger.indy.sdk.wallet;

import org.hyperledger.indy.sdk.ErrorCode;
import org.hyperledger.indy.sdk.IndyException;

/**
 * Exception thrown when attempting to use a wallet with a pool other than the pool the wallet was created for.
 */
public class WrongWalletForPoolException extends IndyException
{
	private static final long serialVersionUID = -8931044806844925321L;
	private final static String message = "The wallet specified is not compatible with the open pool.";

	/**
	 * Initializes a new WrongWalletForPoolException.
	 */
	public WrongWalletForPoolException()
	{
		super(message, ErrorCode.WalletIncompatiblePoolError.value());
	}
}
